packages/graph/hash_graph/lib/graph/src/store/postgres/knowledge/entity/read.rs

```diff
@@ -3,82 +3,72 @@ use chrono::{DateTime, Utc};
 use error_stack::{IntoReport, Result, ResultExt};
 use futures::{StreamExt, TryStreamExt};
 use postgres_types::ToSql;
-use tokio_postgres::GenericClient;
+use tokio_postgres::{GenericClient, RowStream};
 
 use crate::{
-    knowledge::EntityId,
-    ontology::AccountId,
-    store::{crud, AsClient, PostgresStore, QueryError},
-    AllLatest, PersistedEntity, VersionedUri,
+    knowledge::{EntityId, PersistedEntity},
+    ontology::{types::uri::VersionedUri, AccountId},
+    store::{
+        crud,
+        query::{EntityQuery, EntityVersion},
+        AsClient, PostgresStore, QueryError,
+    },
 };
 
-#[async_trait]
-impl<C: AsClient> crud::Read<'_, EntityId, PersistedEntity> for PostgresStore<C> {
-    type Output = PersistedEntity;
-
-    async fn get(&self, identifier: EntityId) -> Result<Self::Output, QueryError> {
-        let row = self
-            .as_client()
-            .query_one(
-                r#"
-                SELECT properties, entity_id, entities.version, ids.base_uri, ids.version, created_by
+async fn by_id_by_latest_version(
+    client: &(impl GenericClient + Sync),
+    entity_id: EntityId,
+) -> Result<RowStream, QueryError> {
+    client
+        .query_raw(
+            r#"
+            SELECT properties, entity_id, entities.version, ids.base_uri, ids.version, created_by
+            FROM entities
+            INNER JOIN ids ON ids.version_id = entities.entity_type_version_id
+            WHERE entity_id = $1 AND entities.version = (
+                SELECT MAX("version")
                 FROM entities
-                INNER JOIN ids ON ids.version_id = entities.entity_type_version_id
-                WHERE entity_id = $1 AND entities.version = (
-                    SELECT MAX("version")
-                    FROM entities
-                    WHERE entity_id = $1
-                );
-                "#,
-                &[&identifier],
-            )
-            .await
-            .into_report()
-            .change_context(QueryError)
-            .attach_printable(identifier)?;
-
-        let entity = serde_json::from_value(row.get(0))
-            .into_report()
-            .change_context(QueryError)?;
-
-        let entity_id: EntityId = row.get(1);
-        let entity_version: DateTime<Utc> = row.get(2);
-        let type_base_uri: String = row.get(3);
-        let type_version: i64 = row.get(4);
-        let created_by: AccountId = row.get(5);
+                WHERE entity_id = $1
+            );
+            "#,
+            &[&entity_id],
```

Comment:
Here `parameter_list` can be utilized as well

Reply:
Applied in eefbbb505466911cf5d6ff78dd0ac2e8945eb546